### PR TITLE
Mob 2965 ui adyen flow tests

### PR DESCRIPTION
**Pull request by jeremyauclert**
## JIRA
[MOB-2965](https://karhoo.atlassian.net/browse/MOB-2965)

## DESCRIPTION
Adyen user flow tests (Booking and trips)

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [x] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: uisdk/src/androidTest/java/com/karhoo/uisdk/trip/adyenTrip/AdyenTripTests.kt [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.runner.RunWith
+import java.net.HttpURLConnection.HTTP_CREATED
+import java.net.HttpURLConnection.HTTP_INTERNAL_ERROR
+import java.net.HttpURLConnection.HTTP_OK
+
+@RunWith(AndroidJUnit4::class)
+class AdyenTripTests : Launch {
```
Comment (jsm20) [inline uisdk/src/androidTest/java/com/karhoo/uisdk/trip/adyenTrip/AdyenTripTests.kt:56]: Not sure we need separate Adyen/Braintree trip tests. Definitely for booking but the trip logic is independent of PSP. @mohadian your thoughts?

Comment (jeremyauclert) [reply uisdk/src/androidTest/java/com/karhoo/uisdk/trip/adyenTrip/AdyenTripTests.kt:56]: I did wonder that at the time...

File: app/src/androidTest/java/com/karhoo/karhootraveller/splash/SplashTests.kt [hunk 2]
```diff
@@ -306,15 +305,15 @@ class SplashTests : Launch {
             reverseGeocodeResponse(HTTP_OK, REVERSE_GEO_SUCCESS)
         }
         splash(this) {
-            sleep(MEDIUM)
+            shortSleep()
```
Comment (mohadian) [inline app/src/androidTest/java/com/karhoo/karhootraveller/splash/SplashTests.kt:308]: should we use `mediumSleep()`

File: uisdk/src/androidTest/java/com/karhoo/uisdk/booking/braintree/BraintreeBookingFlowTests.kt [hunk 3]
```diff
@@ -340,20 +318,20 @@ class BookingFlowTests : Launch {
...
         booking {
-            sleep()
+            shortSleep()
         } result {
-            selectedPickupAddressIsVisible(address = TestData.SELECTED_ADDRESS)
+            selectedPickupAddressIsVisible(address = SELECTED_ADDRESS)
         }
         booking {
-            sleep(MEDIUM)
+            shortSleep()
```
Comment (mohadian) [inline uisdk/src/androidTest/java/com/karhoo/uisdk/booking/braintree/BraintreeBookingFlowTests.kt:330]: mediumSleep()?

File: uisdk/src/androidTest/java/com/karhoo/uisdk/booking/adyen/AdyenBookingTests.kt [hunk 4]
```diff
@@ -0,0 +1,621 @@
...
+     * Then:    I see an error informing me that pick up and drop off cannot be identical
+     **/
+    @Test
+    @AllowFlaky(attempts = 5)
+    fun pickUpAndDropOffAddressesCannotBeTheSame() {
+        serverRobot {
+            reverseGeocodeResponse(HTTP_OK, REVERSE_GEO_SUCCESS)
+            quoteIdResponse(HTTP_BAD_REQUEST, ADDRESSES_IDENTICAL)
+        }
+        booking(this, BraintreeBookingTests.IDENTICAL_ADDRESSES_TRIP_INTENT) {
```
Comment (mohadian) [inline uisdk/src/androidTest/java/com/karhoo/uisdk/booking/adyen/AdyenBookingTests.kt:396]: Should not this be `AdyenBookingTests`

File: app/src/androidTest/java/com/karhoo/karhootraveller/splash/SplashTests.kt (hunk 2 above)
Comment (jeremyauclert) [reply app/src/androidTest/java/com/karhoo/karhootraveller/splash/SplashTests.kt:308]: I've tested it and a short sleep is enough

File: uisdk/src/androidTest/java/com/karhoo/uisdk/booking/braintree/BraintreeBookingFlowTests.kt (hunk 3 above)
Comment (jeremyauclert) [reply uisdk/src/androidTest/java/com/karhoo/uisdk/booking/braintree/BraintreeBookingFlowTests.kt:330]: I've tested it and a short sleep is enough

File: uisdk/src/androidTest/java/com/karhoo/uisdk/booking/adyen/AdyenBookingTests.kt (hunk 4 above)
Comment (jeremyauclert) [reply uisdk/src/androidTest/java/com/karhoo/uisdk/booking/adyen/AdyenBookingTests.kt:396]: Correct, I'll change it now.

**Review by mohadian — APPROVED** [review]

**Pull request merged**